db.go

```diff
@@ -247,34 +247,73 @@ func newDBMetrics(db *DB, r prometheus.Registerer) *dbMetrics {
 
 // DBReadOnly provides APIs for read only operations on a database.
 type DBReadOnly struct {
-	db *DB
+	logger     log.Logger
+	dir        string
+	registerer prometheus.Registerer
```

Comment:
Do you need metrics really? All the CLI jobs are batch jobs.

Reply:
Yeah I have removed it and will add when/if needed.

I use these for tests all the time when I want to check some internal state during the execution, but there is no immediate need for it right now.